### PR TITLE
Replaced missing blowfish dependency

### PR DESCRIPTION
**Pull request by dheid**
Hi Leon!

Please consider the following changes since the BlowfishJ dependency is not resolvable using the Maven Central repository. I forked the original code of Markus Hahn on SourceForge and created a GitHub repository out of it:

https://github.com/dheid/blowfishj

I also added it to the Sonatype repository and now it is available on the Maven central: http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22blowfishj%22

Deprecated code is now removed from the BlowfishJ code. So some changes on the CryptTool were necessary.

Best wishes

Daniel

### CODE REVIEW

**Comment by dheid** [comment]
Hi Leon!

These refactorings are just little improvements, that you can partly read in Effective Java and so on. They don't harm your code and are easy to merge. I chose the ones, that I think are meaningful to deny code smells and provide some more performance. 

Since it takes long time to explain I placed you some really good explanations I found on the web:

- isEmpty (in some data structure implementations very useful): http://stackoverflow.com/a/9341835
- Autoboxing (provides the best method for conversion): https://effective-java.com/2010/05/the-advantages-and-traps-of-autoboxing/

Kind regards
Daniel

**Comment by dheid** [comment]
I did another revision on this and using the provided Java Cryptographic extension. This should be much better now.

File: src/main/java/net/anotheria/util/NameAnalyzer.java [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-import net.anotheria.util.sorter.DummySortType;
-import net.anotheria.util.sorter.IComparable;
-import net.anotheria.util.sorter.StaticQuickSorter;
 /**
  * Simple utility to traverse a directory and analyze the file names.
  * @author lrosenberg
  *
  */
 public class NameAnalyzer {
+
```
Comment (dvayanu) [inline src/main/java/net/anotheria/util/NameAnalyzer.java]: Why would you replace System.outs with log.infos in a command line utility?

File: src/main/java/net/anotheria/util/NameAnalyzer.java [hunk 2]
```diff
@@ -32,32 +36,26 @@
...
 	 */
-	public static void main(String a[]) throws IOException{
-		//test();
-		///*
+	public static void main(String... a) {
 		proceed(new File("."));
 		
-		System.out.println("Finished found files: "+filesFound+", dirs: "+dirFound+", java files: "+javaFilesFound);
-		System.out.println("Found "+names.size()+" names, "+elements.size()+" elements.");
+		log.info("Finished found files: "+filesFound+", dirs: "+dirFound+", java files: "+javaFilesFound);
```
Comment (dvayanu) [inline src/main/java/net/anotheria/util/NameAnalyzer.java]: Why would you replace System.outs with log.infos in a command line utility?

File: src/main/java/net/anotheria/util/concurrency/SafeIdBasedLockManager.java [hunk 3]
```diff
@@ -10,26 +10,32 @@
...
-    private Map<K, IdBasedLock<K>> locks = new HashMap<K, IdBasedLock<K>>();
+    private Map<K, IdBasedLock<K>> locks = new HashMap<>();
 
-    public synchronized IdBasedLock<K> obtainLock(K id) {
-        IdBasedLock<K> lock = locks.get(id);
-        if (lock != null) {
-            lock.increaseRefCount();
+    @Override
+    public IdBasedLock<K> obtainLock(K id) {
+        synchronized (this) {
```
Comment (dvayanu) [inline src/main/java/net/anotheria/util/concurrency/SafeIdBasedLockManager.java]: Why did you add a synchronized block around this part? This is breaking all the functionality!

**Comment by dvayanu** [comment]
One more comment, could you in the future provide smaller pull requests? Its really hard to review 125 files... especially if some of the changes are ... questionable (I added some commends)... for example using of logging frameworks instead of sys-out in command line tools is ... inconvenient and uncommon... 
Also did you performed any checks that the jce solution for blowfish is able to decrypt text previously decrypted with blowfish?

**Comment by dvayanu** [comment]
Can you merge net.anotheria.util.crypt.CryptToolCompatibilityTest into your branch and tell us if it still runs ;-) This would make me happier with backward compatibility of the crypttool. We have millions of encrypted tokens in field, would be unpleasant if they all go unuseable.

**Comment by dheid** [comment]
I merged the changes into the branch and everything still runs. :-)

`Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.072 sec - in net.anotheria.util.crypt.CryptToolCompatibilityTest
Running net.anotheria.util.crypt.MD5Test
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec - in net.anotheria.util.crypt.MD5Test
Running net.anotheria.util.crypt.CryptToolTest
Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.022 sec - in net.anotheria.util.crypt.CryptToolTest
Running net.anotheria.util.resource.ResourceTest`

**Comment by dvayanu** [comment]
I will merge pull request now and review remaining changes locally.
